### PR TITLE
feat: add stop delay arg

### PR DESCRIPTION
**Pull request by RafaelGSS**
Address: https://github.com/clinicjs/node-clinic-doctor/issues/362

### CODE REVIEW

File: test/cli-doctor-stop-delay.test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }, function (err) {
+      t.error(err)
+      t.end()
+    })
+  })
+})
+
+// TODO: skipping until the above statement is fulfilled
+test('clinic doctor --stop-delay --on-port - exceeding timeout', { skip: true }, function (t) {
+  // TODO: rafaelgss
```
Comment (RafaelGSS) [inline test/cli-doctor-stop-delay.test.js:44]: @mcollina any idea how to make it work?

Comment (mcollina) [reply test/cli-doctor-stop-delay.test.js:44]: why it's not working?

Comment (RafaelGSS) [reply test/cli-doctor-stop-delay.test.js:44]: `node-tap` doesn't have an API for an expected timeout. When a `timeout` is raised it automatically mark the test as `not ok`.

Comment (mcollina) [reply test/cli-doctor-stop-delay.test.js:44]: What's wrong in using setTimeout?

**Comment by RafaelGSS** [comment]
ping @mcollina

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**